modules/web/client/src/main/scala/observe/ui/components/MainApp.scala

```diff
@@ -121,12 +139,37 @@ object MainApp:
       case WSFrame.Text(text, _) => fs2.Stream(decode[ClientEvent](text))
       case _                     => fs2.Stream.empty
 
-  // TODO IN CRYSTAL: It would be nice to have a .useEffectWhenDepReady instead of doing .void in dependencies!
+  def processStreamEvent(
+    environment:        View[Pot[Environment]],
+    rootModelData:      View[RootModelData],
+    isSynced:           View[Boolean],
```

Comment:
Maybe some `NewType`s insted of plain `boolean`?

Reply:
Good idea